test/test-parse.cpp

```diff
@@ -1826,5 +1837,71 @@ TEST(test_parse, debugger_statement) {
                 ElementsAre(spy_visitor::visited_variable_use{u8"x"}));
   }
 }
+
+TEST(test_parse, report_missing_semicolon_for_declarations) {
+  {
+    spy_visitor v;
+    padded_string code(u8"let x = 2\nfor (;;) { console.log(); }");
+    parser p(&code, &v);
+    p.parse_and_visit_statement(v);
+    p.parse_and_visit_statement(v);
+    EXPECT_THAT(v.variable_declarations,
+                ElementsAre(spy_visitor::visited_variable_declaration{
+                    u8"x", variable_kind::_let}));
+    EXPECT_THAT(v.variable_uses,
+                ElementsAre(spy_visitor::visited_variable_use{u8"console"}));
+    EXPECT_THAT(v.errors, IsEmpty());
+  }
```

Comment:
Minor: The test talks about *reporting* missing semicolons, but these lines show that QLJS should *not* report a missing semicolon in this case. Move this code to a different test, or rename the test.